Automattic/newspack-plugin
# Analytics events framework

### All Submissions:






### Changes proposed in this Pull Request:



Supersedes #524 

This PR is like #524 but without the UI parts (which are probably unnecessary anyways). 
 
**Discussion**: Do we want to make these events [non-interaction events](https://support.google.com/analytics/answer/1033068#NonInteractionEvents)? I think this is what was causing issues with bounce rates on some sites, because it doesn't count as a bounce in GA if an event is sent. E.g. if a user scrolls down 25% of the way through an article then leaves or gets shown a popup then leaves, my understanding is it won't count it as a bounce since an interaction event fired. Thoughts?

### How to test the changes in this Pull Request:

1. Connect Site Kit and analytics.
2. In AMP mode, scroll down an article and click on social buttons. You should see events logged in GA.
3. In non-AMP mode, scroll down an article and click on social buttons. You should see events logged in GA. (In my testing I don't though for some reason. Everything is firing and sending the event correctly as far as I can tell, though. Even manual calls to `ga( 'send', 'foo', 'bar', 'baz' );` in the dev console don't show any events for me)

<img width="1296" alt="Screen Shot 2020-05-07 at 11 59 07 AM" src="https://user-images.githubusercontent.com/7317227/81334679-4149ca00-905b-11ea-97ca-c95b56672a7e.png">

### Other information:
